src/nvim/os/env.c

```diff
@@ -141,6 +141,28 @@ void init_homedir(void)
 
   char_u *var = (char_u *)os_getenv("HOME");
 
+#ifdef WIN32
+  // Typically, $HOME is not defined on Windows, unless the user has
+  // specifically defined it for Vim's sake. However, on Windows NT
+  // platforms, $HOMEDRIVE and $HOMEPATH are automatically defined for
+  // each user. Try constructing $HOME from these.
+  if (var == NULL) {
+    const char *homedrive = os_getenv("HOMEDRIVE");
+    const char *homepath = os_getenv("HOMEPATH");
+    if (homepath == NULL) {
+        homepath = "\\";
+    }
+    if (homedrive != NULL && strlen(homedrive) + strlen(homepath) < MAXPATHL) {
+      snprintf((char *)NameBuff, MAXPATHL, "%s%s", homedrive, homepath);
+      if (NameBuff[0] != NUL) {
+        var = NameBuff;
+        // Also set $HOME, it's needed for .viminfo.
```

Comment:
Replacing `_viminfo` with `.viminfo` is useless and incorrect: *both* names are no longer used because viminfo was replaced with ShaDa with different naming convention (`main.shada`, `main.shada.tmp.{char}`, XDG-compliant location).